### DISCUSSION TITLE
"offset: 1" leads to leading zeros in Contents. How to fix it?

### DISCUSSION THREAD

**Discussion by johnmapeson** [typst/typst#4941] [Q&A] [answered]
`doc.typ`:

```
#heading(level: 1, outlined: false)[My document]

#outline()
#pagebreak()

#set page(numbering: "1")
#counter(page).update(2)
#set heading(offset: 1, numbering: "1.")

#include "part1.typ"
#include "part2.typ"
```

`part1.typ`:

```
= Lorem ipsum

#lorem(50)
```

`part2.typ`:

```
= Dolor sit

#lorem(50)
```

I want "Lorem ipsum" and "Dolor sit" to be level-2 headings in the combined document, and this is why I added `offset: 1`:

```
#set heading(offset: 1, numbering: "1.")
```

But then outline, that is, Contents, looks weird:

```
Contents
0.1. Lorem ispum ...
0.2. Dolo sit ......
```

Why is this? How to make numbers there just `1.` and `2.`, not `0.1.` and `0.2.`?

**Comment by PgBiel** [comment]
They are level 2 now as you said, and level 2 headings have 2 numbers, level 3 headings have 3 numbers, and so on. There is a level 1 heading before them, but it isn't numbered, so the level 1 number is 0 at that point (not numbered).

If I have read your intentions correctly, you're trying to write a document title at the top. If so, you should rather use large text instead (e.g. `#let title(contents) = align(center, text(54pt)[#contents])` and then `#title[My document]`) and have your headings be level 1 instead of level 2 (remove the offset).

Usually the offset is used when you have level 1 headings for chapters whose contents are separate files which are later `#include`d into a main file. Then you can have level 1 in the subfile but level 2 in the final document through `offset`. This doesn't seem to be what you're trying to do here.

**Reply by johnmapeson** [reply]
> Usually the offset is used when you have level 1 headings for chapters whose contents are separate files which are later `#include`d into a main file. Then you can have level 1 in the subfile but level 2 in the final document through `offset`. 

It sounds exactly what I'm trying to do :) But why this  doesn't work? Could you show a correct example?

**Reply by PgBiel** [reply]
In this case you can either number your chapters and have them appear in your TOC as well, or you can customize the numbering format so it ignores the first number for non-chapters:

```typ
#set heading(numbering: (..n) => if n.pos().len() == 1 { numbering("1.", ..n) } else { numbering("1.", ..n.pos().slice(1)) })

**Reply by johnmapeson** [reply]
Thanks a lot. But there are two things that I don't understand.

- You said that _usually the offset is used when you have level 1 headings for chapters whose contents are separate files which are later #included into a main file, so I can have level 1 in the subfile but level 2 in the final document through offset._ Does it mean that if I don't use code like the one you have posted, the described scenario assumes that headings don't have automatic numbering?

- And are you sure the code you have posted works correctly? I tried it as shown below, and "Lorem ipsum" and "Dolor sit" headings in the combined file seems to be level-1 ones, not level-2.

  ```
  #heading(level: 1, outlined: false)[My document]
  
  #outline()
  #pagebreak()
  
  #set page(numbering: "1")
  #counter(page).update(2)
  #set heading(numbering: (..n) => if n.pos().len() == 1 { numbering("1.", ..n) } else { numbering("1.", ..n.pos().slice(1)) })
  #show heading.where(level: 1): set text(red)
  #show heading.where(level: 2): set text(blue)
  
  #include "part1.txt"
  #include "part2.txt"
  ```

**Reply by sitandr** [reply, marked as answer]
@PgBiel implied that you would still need to keep `set offset: 1`, so you would either need to add it to the rule too or use just another `set`. The whole line that you should use then looks like this:

```typ
#set heading(offset: 1, numbering: (..n) => if n.pos().len() == 1 { numbering("1.", ..n) } else { numbering("1.", ..n.pos().slice(1)) })
```

I've checked that your code then compiles well, the last two headings are blue.
